### PR TITLE
Initial draft for observation mapping

### PR DESCRIPTION
**Pull request by brandantck**
1. Add logging config
2. Add initial code for observation mapping

TODO: 
1. Add generation of running number as observation_id after data mapping has been completed.
2. Handle mapping of duplicate source_code values in source_to_concept_mapping table
3. Update mapping logic for visit_occurence_id
4. Double check hardcoded mapping values for observation_concept_id at file:`etl/pypasar/omop/observation_utils/config.py`, variable: `observation_concept_id_specific_config`

### CODE REVIEW

File: etl/pypasar/omop/observation.py [hunk 1]
```diff
@@ -11,26 +20,169 @@ class observation:
...
+                # Add a column to indicate which source row is from
+                temp[SOURCE_TABLE_COL_NAME] = source
+                df = pd.concat([df, temp], ignore_index=True)
+        return df
+
+    def get_omop_person_table(self) -> pd.DataFrame:
+        with self.engine.connect() as connection:
+            df = pd.read_sql(
+                text(
+                    f"""SELECT person_id, person_source_value from {os.getenv("POSTGRES_OMOP_SCHEMA")}.person;"""
```
Comment (satish-a0) [inline etl/pypasar/omop/observation.py:174]: NIT: Would recommend handling scalability by avoiding loading the full table.

File: etl/pypasar/omop/observation_utils/config.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        "omop": "value_as_string"
+    }
+
+    observation_type_concept_id_mapping = {
+        "pasar": "",
+        "omop": "observation_type_concept_id"
+    }
+
+    visit_occurrence_id_mapping = {
+        "pasar": "session_id",
```
Comment (satish-a0) [inline etl/pypasar/omop/observation_utils/config.py:37]: To update logic

File: etl/pypasar/omop/observation_utils/mappings.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+class ObservationMappings():
+    @mapping_wrapper
+    def map_observation_id(df: pd.DataFrame) -> pd.DataFrame:
+        observation_id_mapping = ObservationMappingConfig.observation_id_mapping
+        # Sort and reset index to set dataframe index as running index
+        df = df.sort_values(observation_id_mapping["pasar"], ascending=[
+                            False, False]).reset_index(drop=True)
+
+        # reset_index again to get id as a column
+        df = df.reset_index(names=observation_id_mapping["omop"])
```
[the remark this answers is not in the document]
Comment (brandantck) [reply etl/pypasar/omop/observation_utils/mappings.py:35]: Okay will do that
https://github.com/satish-a0/pasar/pull/8/commits/72112471c1ffff0fbf72ff30ea4a4cb82f8d423e

File: etl/pypasar/omop/observation_utils/config.py (hunk 2 above)
Comment (brandantck) [reply etl/pypasar/omop/observation_utils/config.py:37]: Can confirm what needs to be updated for this mapping?
From the [document](https://github.com/sidataplus/PASAR_ETL_Spec/blob/main/observation.md), it seems like visit_occurrence_id is just a simple mapping to session_id

Comment (satish-a0) [reply etl/pypasar/omop/observation_utils/config.py]: Mentioned [here](https://teams.microsoft.com/l/message/19:cf1d1c096fb14b7fbcef22f547c33211@thread.v2/1727136341596?context=%7B%22contextType%22%3A%22chat%22%7D)

**Review by satish-a0 — APPROVED** [review]

**Pull request merged**